src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String SEPARATOR = " ";
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder finalReport = new StringBuilder();
+        String[] checkLine;
+        boolean isDateInRange;
+        LocalDate dateRangeBegin;
+        LocalDate dateRangeEnd;
+        LocalDate dataSalary;
+
+        finalReport.append("Report for period ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo);
+
+        for (String name : names) {
+            finalReport.append(System.lineSeparator()).append(name).append(" - ");
```

Comment:
can be combined with row 40 into one command to make all appends with salary at once